File: 1erdepartamental/c++/13.tipo_dato_puntero/main.cpp
#include <iostream>
#include "core/EstadisticasEstudiantes.h"

int main() {
    // Crea estadisticas
    EstadisticasEstudiantes stats;
    
    // Agrega estudiantes
    stats.agregar(Estudiante("Ana", 9.5));
    stats.agregar(Estudiante("Luis", 8.7));
    stats.agregar(Estudiante("Carlos", 9.8));
    
    std::cout << "Promedio general: " << stats.promedioGeneral() << std::endl;
    std::cout << "Mejor estudiante: " << stats.mejorEstudiante() << std::endl;
    
    return 0;
}

File: 1erdepartamental/python/13.tipo_dato_puntero/main.py
"""
Programa: Tipo Dato Puntero
Demuestra puntero tipado con objetos personalizados
"""

from core.estadisticas import EstadisticasEstudiantes, Estudiante

if __name__ == "__main__":
    stats = EstadisticasEstudiantes()

    stats.agregar(Estudiante("Ana", 9.5))
    stats.agregar(Estudiante("Luis", 8.7))
    stats.agregar(Estudiante("Carlos", 9.8))

    print(f"Promedio general: {stats.promedio_general():.2f}")
    print(f"Mejor estudiante: {stats.mejor_estudiante()}")
